internal/charmstore/store.go

```diff
@@ -649,10 +649,15 @@ func (s *Store) UpdateBaseEntity(url *router.ResolvedURL, update interface{}) er
 // An error is returned if no channels are provided. For the time being,
 // the only supported channels are "development" and "stable".
 func (s *Store) Publish(url *router.ResolvedURL, channels ...Channel) error {
+	var updateSearch bool
 	// Validate channels.
 	actual := make([]Channel, 0, len(channels))
 	for _, c := range channels {
-		if c == DevelopmentChannel || c == StableChannel {
+		switch c {
```

Comment:
Maybe just trust that the Channel values have been validated for correctness before calling this function.